sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/CollectionExpressionsSuite.scala

```diff
@@ -2250,6 +2250,77 @@ class CollectionExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper
       Seq(2d))
   }
 
+  test("Array Insert") {
+    val a1 = Literal.create(Seq(1, 2, 4), ArrayType(IntegerType))
+    val a2 = Literal.create(Seq(1, 2, null, 4, 5, null), ArrayType(IntegerType))
+    val a3 = Literal.create(Seq[Boolean](true, false, true), ArrayType(BooleanType))
+    val a4 = Literal.create(Seq[Byte](1, 2, 3, 2), ArrayType(ByteType))
+    val a7 = Literal.create(Seq[Short](1, 2, 3, 2), ArrayType(ShortType))
+    val a9 = Literal.create(Seq[Float](1.1F, 2.2F, 3.3F, 2.2F), ArrayType(FloatType))
+    val a11 = Literal.create(Seq[Double](1.1, 2.2, 3.3, 2.2), ArrayType(DoubleType))
+    val a13 = Literal.create(Seq(1L, 2L, 4L), ArrayType(LongType))
+    val a15 = Literal.create(Seq("b", "a", "c"), ArrayType(StringType, false))
+    val a16 = Literal.create(Seq("b", null, "a", "g", null), ArrayType(StringType, true))
+    val a18 = Literal.create(null, ArrayType(StringType))
+
+    val litTwoInt = Literal.create(2, IntegerType)
+    val litThreeInt = Literal.create(3, IntegerType)
+    val litOneInt = Literal.create(1, IntegerType)
+    val litFourInt = Literal.create(4, IntegerType)
+    val litNullInt = Literal.create(null, IntegerType)
+    val litZeroInt = Literal.create(0, IntegerType)
+    val litTenInt = Literal.create(10, IntegerType)
+    val litMinusTwoInt = Literal.create(-2, IntegerType)
+    val litMinusTenInt = Literal.create(-10, IntegerType)
+    val litThreeLong = Literal.create(3L, LongType)
+    val litBoolTrue = Literal.create(true, BooleanType)
+    val litThreeByte = Literal.create(5.asInstanceOf[Byte], ByteType)
+    val litThreeShort = Literal.create(3.asInstanceOf[Short], ShortType)
+    val litFourFourFloat = Literal.create(4.4F, FloatType)
+    val litFourFourDouble = Literal.create(4.4, DoubleType)
+    val litDString = Literal.create("d", StringType)
+
+    // basic additions per type
+    checkEvaluation(ArrayInsert(a1, litTwoInt, litThreeInt), Seq(1, 2, 3, 4))
```

Comment:
what about directly using the Literal expressions for `pos` and `value` ?

e.g.

```suggestion
    checkEvaluation(ArrayInsert(a1, Literal(2), Literal.create(3, IntegerType)), Seq(1, 2, 3, 4))
```